### PR TITLE
FAT-1205:C163928 Test acquisition unit restrictions for Fund records

### PR DESCRIPTION
**Pull request by IhorBohdan**
https://foliotest.testrail.io/index.php?/cases/view/163928
https://issues.folio.org/browse/FAT-1205

### CODE REVIEW

File: cypress/e2e/finance/funds/test-acquisition-unit-restrictions-for-funds.cy.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import FiscalYears from '../../../support/fragments/finance/fiscalYears/fiscalYears';
+import TopMenu from '../../../support/fragments/topMenu';
+import Ledgers from '../../../support/fragments/finance/ledgers/ledgers';
+import Users from '../../../support/fragments/users/users';
+import Funds from '../../../support/fragments/finance/funds/funds';
+import FinanceHelp from '../../../support/fragments/finance/financeHelper';
+import SettingsMenu from '../../../support/fragments/settingsMenu';
+import AcquisitionUnits from '../../../support/fragments/settings/acquisitionUnits/acquisitionUnits';
+
+describe('ui-finance: Transactions', () => {
```
Comment (penfriend) [inline cypress/e2e/finance/funds/test-acquisition-unit-restrictions-for-funds.cy.js:13]: the test fails when running on the cypress env

Comment (penfriend) [reply cypress/e2e/finance/funds/test-acquisition-unit-restrictions-for-funds.cy.js:13]: image

File: cypress/e2e/finance/funds/test-acquisition-unit-restrictions-for-funds.cy.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      .then(userProperties => {
+        user = userProperties;
+      });
+  });
+
+  after(() => {
+    cy.loginAsAdmin({ path:SettingsMenu.acquisitionUnitsPath, waiter: AcquisitionUnits.waitLoading });
+    AcquisitionUnits.unAssignAdmin(defaultAcquisitionUnit.name);
+    AcquisitionUnits.delete(defaultAcquisitionUnit.name);
+    cy.loginAsAdmin({ path:TopMenu.fundPath, waiter: Funds.waitLoading });
```
Comment (penfriend) [inline cypress/e2e/finance/funds/test-acquisition-unit-restrictions-for-funds.cy.js:67]: there is no need to login as admin one more time. you can use cy.visit and wait while the page is loaded

File: cypress/e2e/finance/funds/test-acquisition-unit-restrictions-for-funds.cy.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    Funds.createFundWithAU(defaultfund, defaultLedger, defaultAcquisitionUnit.name);
+    cy.loginAsAdmin({ path:SettingsMenu.acquisitionUnitsPath, waiter: AcquisitionUnits.waitLoading });
+    AcquisitionUnits.unAssignUser(defaultAcquisitionUnit.name);
+    cy.login(user.username, user.password, { path:TopMenu.fundPath, waiter: Funds.waitLoading });
+    FinanceHelp.searchByName(defaultfund.name);
+    Funds.checkZeroSearchResultsHeader();
+    cy.loginAsAdmin({ path:SettingsMenu.acquisitionUnitsPath, waiter: AcquisitionUnits.waitLoading });
+    AcquisitionUnits.edit(defaultAcquisitionUnit.name);
+    AcquisitionUnits.selectViewCheckbox();
+    cy.pause();
```
Comment (penfriend) [inline cypress/e2e/finance/funds/test-acquisition-unit-restrictions-for-funds.cy.js:94]: why do you need this method?

File: cypress/e2e/finance/funds/test-acquisition-unit-restrictions-for-funds.cy.js (hunk 2 above)
Comment (IhorBohdan) [reply cypress/e2e/finance/funds/test-acquisition-unit-restrictions-for-funds.cy.js]: Removed

File: cypress/support/fragments/settings/acquisitionUnits/acquisitionUnits.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+  fillInInfo: (name) => {
+    cy.do([
+      TextField({ name: 'name' }).fillIn(name),
+      viewCheckbox.click(),
+      saveAUButton.click(),
+      findUserButton.click(),
+      userSearchModal.find(searchTextField).fillIn('diku'),
+      Button('Search').click(),
+      firstSearchResult.find(Checkbox()).click(),
```
[the remark this answers is not in the document]
Comment (IhorBohdan) [reply cypress/support/fragments/settings/acquisitionUnits/acquisitionUnits.js]: Done

File: cypress/support/fragments/settings/acquisitionUnits/acquisitionUnits.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  },
+
+  fillInInfo: (name) => {
+    cy.do([
+      TextField({ name: 'name' }).fillIn(name),
+      viewCheckbox.click(),
+      saveAUButton.click(),
+      findUserButton.click(),
+      userSearchModal.find(searchTextField).fillIn('diku'),
+      Button('Search').click(),
```
[the remark this answers is not in the document]
Comment (IhorBohdan) [reply cypress/support/fragments/settings/acquisitionUnits/acquisitionUnits.js]: Done

File: cypress/support/fragments/settings/acquisitionUnits/acquisitionUnits.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  unAssignUser: (AUName) => {
+    cy.do([
+      auListPane.find(Button(AUName)).click(),
+      assignedUsersSection.find(MultiColumnListCell({ row: 1, columnIndex: 2 })).find(trashButton).click()
+    ]);
+  },
+
+  unAssignAdmin: (AUName) => {
+    cy.do([
+      auListPane.find(Button(AUName)).click(),
```
[the remark this answers is not in the document]
Comment (penfriend) [reply cypress/support/fragments/settings/acquisitionUnits/acquisitionUnits.js:65]: sorry, you are right.

File: cypress/e2e/finance/funds/test-acquisition-unit-restrictions-for-funds.cy.js (hunk 1 above)
Comment (penfriend) [reply cypress/e2e/finance/funds/test-acquisition-unit-restrictions-for-funds.cy.js]: Please create a ticket for this issue

**Review by penfriend — APPROVED** [review]

**Review by MaksimTerekhov — APPROVED** [review]

**Pull request merged**